packages/protocol/contracts/L2/CrossChainOwned.sol

```diff
@@ -24,23 +24,24 @@ import "../bridge/IBridge.sol";
 /// signals for transaction approval.
 /// @dev Notice that when sending the message on the owner chain, the gas limit of the message must
 /// not be zero, so on this chain, some EOA can help execute this transaction.
-abstract contract CrossChainOwned is EssentialContract {
+abstract contract CrossChainOwned is EssentialContract, IMessageInvocable {
     uint64 public ownerChainId; // slot 1
     uint64 public nextTxId;
     uint256[49] private __gap;
 
     event TransactionExecuted(uint64 indexed txId, bytes4 indexed selector);
 
-    error XCO_INVALID_TX_ID();
     error XCO_INVALID_OWNER_CHAINID();
+    error XCO_INVALID_TX_ID();
     error XCO_PERMISSION_DENIED();
     error XCO_TX_REVERTED();
 
-    function executeCrossChainTransaction(uint64 txId, bytes calldata txdata) external {
-        if (txId != nextTxId) revert XCO_INVALID_TX_ID();
-
+    function onMessageInvocation(bytes calldata data) external payable nonReentrant whenNotPaused {
```

Comment:
I don't think this one can be `nonReentrant` because this contract needs to call into other `nonReentrant` functions.